auraed/src/observe/cgroup_cache.rs

```diff
@@ -0,0 +1,130 @@
+ *          Apache 2.0 License Copyright © 2022-2023 The Aurae Authors        *
+ *                                                                            *
+ *                +--------------------------------------------+              *
+ *                |   █████╗ ██╗   ██╗██████╗  █████╗ ███████╗ |              *
+ *                |  ██╔══██╗██║   ██║██╔══██╗██╔══██╗██╔════╝ |              *
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::fs;
+use std::os::unix::prelude::DirEntryExt;
+use tracing::warn;
+
+/// Used for looking up cgroup paths by inode number
+struct CgroupCache {
+    root: OsString,
+    cache: HashMap<u64, OsString>,
+}
+
+impl CgroupCache {
+    pub fn new(root: OsString) -> Self {
+        Self { root, cache: HashMap::new() }
+    }
+
+    pub fn get(&mut self, ino: u64) -> Option<OsString> {
+        if let Some(path) = self.cache.get(&ino) {
+            Some(path.clone())
+        } else {
+            self.refresh_cache();
+            self.cache.get(&ino).cloned()
+        }
+    }
+
+    fn refresh_cache(&mut self) {
+        fs::read_dir(&self.root)
```

Comment:
We have nested cgroups, does this read those directories too? Docs don't make it seem like it. I think we already have a dep on https://crates.io/crates/walkdir which may be helpful